src/adapter-types/src/bootstrap_builtin_cluster_config.rs

```diff
@@ -0,0 +1,22 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+//! Types for bootstrap builtin cluster configuration.
+
+#[derive(Debug, Clone)]
+pub struct BootstrapBuiltinClusterConfig {
+    pub size: String,
+    pub replication_factor: u32,
+}
+
+pub const SYSTEM_CLUSTER_DEFAULT_REPLICATION_FACTOR: u32 = 1;
+pub const CATALOG_SERVER_CLUSTER_DEFAULT_REPLICATION_FACTOR: u32 = 1;
+pub const PROBE_CLUSTER_DEFAULT_REPLICATION_FACTOR: u32 = 1;
+pub const SUPPORT_CLUSTER_DEFAULT_REPLICATION_FACTOR: u32 = 0;
+pub const ANALYTICS_CLUSTER_DEFAULT_REPLICATION_FACTOR: u32 = 0;
```

Comment:
While you're here, can you add a comment explaining why these have a default of 0? e.g. that they are ephemeral clusters we spin up only to scrape analytics or for support when debugging

Reply:
sgtm!